File: backend/src/main/java/org/globe42/web/users/PasswordGenerator.java
package org.globe42.web.users;

import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;

import org.springframework.stereotype.Component;

/**
 * A random password generator
 * @author JB Nizet
 */
@Component
public class PasswordGenerator {

    /**
     * The size of generated passwords
     */
    private static final int SIZE = 8;

    /**
     * The characters composing generated passwords. The generators picks randomly in these characters.
     */
    private static final char[] CHARACTERS =
        "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_$%?/+=.<>#*".toCharArray();

    private SecureRandom random;

    public PasswordGenerator() {
        try {
            random = SecureRandom.getInstance("SHA1PRNG");
        }
        catch (NoSuchAlgorithmException e) {
            throw new IllegalStateException(e);
        }
    }

    public String generatePassword() {
        StringBuilder builder = new StringBuilder(SIZE);
        for (int i = 0; i < SIZE; i++) {
            builder.append(CHARACTERS[random.nextInt(CHARACTERS.length)]);
        }
        return builder.toString();
    }
}

File: backend/src/main/kotlin/org/globe42/web/users/PasswordGenerator.kt
package org.globe42.web.users

import org.springframework.stereotype.Component
import java.security.SecureRandom

/**
 * The size of generated passwords
 */
private val SIZE = 8

/**
 * The characters composing generated passwords. The generators picks randomly in these characters.
 */
private val CHARACTERS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_$%?/+=.<>#*".toCharArray()

/**
 * A random password generator
 * @author JB Nizet
 */
@Component
class PasswordGenerator {

    private val random: SecureRandom = SecureRandom.getInstance("SHA1PRNG")

    fun generatePassword(): String {
        val builder = StringBuilder(SIZE)
        for (i in 0 until SIZE) {
            builder.append(CHARACTERS[random.nextInt(CHARACTERS.size)])
        }
        return builder.toString()
    }
}
